src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/MultiDimensionalExtension.java

```diff
@@ -0,0 +1,231 @@
+                try {
+                    executeDescribeDomainsOperation(conveyor);
+                } catch (Exception exception) {
+                    LOGGER.log(Level.SEVERE, "Error executing describe domains operation.", exception);
+                    throw new OWSException(500, "NoApplicableCode", "",
+                            "Error executing describe domains operation:" + exception.getMessage());
+                }
+                break;
+            case GET_HISTOGRAM:
+                try {
+                    executeGetHistogramOperation(conveyor);
+                } catch (Exception exception) {
+                    LOGGER.log(Level.SEVERE, "Error executing get histogram operation.", exception);
+                    throw new OWSException(500, "NoApplicableCode", "",
+                            "Error executing get histogram operation:" + exception.getMessage());
+                }
+                break;
+            case GET_FEATURE:
+                try {
+                    executeGetFeatureOperation(conveyor);
+                } catch (Exception exception) {
+                    LOGGER.log(Level.SEVERE, "Error executing get feature operation.", exception);
+                    throw new OWSException(500, "NoApplicableCode", "",
+                            "Error executing get feature operation:" + exception.getMessage());
+                }
+                break;
+            default:
+                return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void encodeLayer(XMLBuilder xmlBuilder, TileLayer tileLayer) throws IOException {
+        LayerInfo layerInfo = getLayerInfo(tileLayer, tileLayer.getName());
+        if (layerInfo == null) {
+            // dimension are not supported for this layer (maybe is a layer group)
+            return;
+        }
+        List<Dimension> dimensions = DimensionsUtils.extractDimensions(wms, layerInfo);
+        encodeLayerDimensions(xmlBuilder, dimensions);
+    }
+
+    private Domains getDomains(SimpleConveyor conveyor) throws Exception {
+        // getting and parsing the mandatory parameters
+        String layerName = (String) conveyor.getParameter("layer", true);
+        TileLayer tileLayer = tileLayerDispatcher.getTileLayer(layerName);
+        LayerInfo layerInfo = getLayerInfo(tileLayer, layerName);
+        // getting this layer dimensions along with its values
+        List<Dimension> dimensions = DimensionsUtils.extractDimensions(wms, layerInfo);
+        // let's see if we have a spatial limitation
+        ReferencedEnvelope boundingBox = (ReferencedEnvelope) conveyor.getParameter("bbox", false);
+        // add any domain provided restriction and set the bounding box
+        Filter filter = Filter.INCLUDE;
+        for (Dimension dimension : dimensions) {
+            Object restriction = conveyor.getParameter(dimension.getDimensionName(), false);
+            dimension.setBoundingBox(boundingBox);
```

Comment:
This way the DescribeDomains response contains the spatial domain only if it was requested. For vector feature and structured coverage grid readers it should be present regardless instead, and return only the bounds of the remaining features/granules (e.g., if I set the date to a range of few hours, the bbox for scattered satellite images could be pretty small). Anyways, I'd treat this one as a bug and fix it after the pull request has been merged.